File: 1502-can-make-arithmetic-progression-from-sequence/1502-can-make-arithmetic-progression-from-sequence.java
// import java.util.Arrays;

class Solution {
    public boolean canMakeArithmeticProgression(int[] arr) {
        
        
        if (arr.length == 2) {
            return true;
        }
        
        Arrays.sort(arr);
        
        int diff = Math.abs(arr[0] - arr[1]);
        
        for (int i = 2; i < arr.length; i++) {
            if (Math.abs(arr[i] - arr[i - 1]) != diff) {
                return false;
            }
        }
        return true;
        
    }
}

File: 1502-can-make-arithmetic-progression-from-sequence/1502-can-make-arithmetic-progression-from-sequence.py
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        if len(arr) == 2:
            return True
        
        arr.sort()
        diff = abs(arr[0] - arr[1])
        
        for i in range(2, len(arr)):
            if diff != abs(arr[i] - arr[i - 1]):
                return False
    
        return True
        